src/api/uploads/UploadReachability.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * @flow
+ * @file Helper for Box uploads reachability test
+ * @author Box
+ */
+
+import Base from '../Base';
+import LocalStore from '../../util/LocalStore';
+import { DEFAULT_HOSTNAME_UPLOAD } from '../../constants';
+import '../../util/regenerator-runtime';
+
+const CACHED_RESULTS_LOCAL_STORE_KEY = 'uploads-reachability-cached-results';
+
+class UploadsReachability extends Base {
+    localStore: LocalStore;
+
+    /**
+     * [constructor]
+     *
+     * @param {Object} [options]
+     * @param {string} [options.token] - Auth token
+     * @param {string} [options.uploadHost] - Upload host name
```

Comment:
Why do you need to pass in uploadHost in constructor?  Doesn't seem like a necessary part of the state of this module.  To me it makes more sense to pass in uploadHost to a method of this class which then checks whether that host is reachable.

Reply:
makes sense, I'll change that. I was trying to use `Base` class, but it seems unnecessary as I can just create an xhr instance in this class and I'm not using other `Base` methods.